src/implementations/twig/components/button/button.story.js

```diff
@@ -11,117 +9,178 @@ import dataCall from '@ecl/specs-component-button/demo/data--call';
 import dataGhost from '@ecl/specs-component-button/demo/data--ghost';
 import dataSearch from '@ecl/specs-component-button/demo/data--search';
 
+import defaultSprite from '@ecl/resources-ec-icons/dist/sprites/icons.svg';
 import button from './button.html.twig';
 import notes from './README.md';
 
 const iconsList = {};
-iconsList.none = '';
-
 uiIcons.forEach((icon) => {
   iconsList[icon] = icon;
 });
 
-// Preserve the adapted specs.
-const prepareButton = (data, type) => {
-  const typeLabel = type ? tabLabels.required : tabLabels.optional;
-  data.disabled = boolean('disabled', data.disabled, tabLabels.states);
-  data.label = text('label', data.label, tabLabels.required);
-
-  data.variant = select(
-    'variant',
-    [data.variant],
-    data.variant,
-    tabLabels.required
-  );
-
-  data.type = select('type', [data.type], data.type, typeLabel);
+const getArgTypes = () => {
+  const argTypes = {};
+  argTypes.label = {
+    name: 'label',
+    type: { name: 'string', required: true },
+    description: 'The main label of the button',
+    table: {
+      type: { summary: 'string' },
+      defaultValue: { summary: '' },
+      category: 'Content',
+    },
+    control: {
+      type: 'text',
+    },
+  };
+  argTypes.icon_name = {
+    name: 'icon name',
+    type: { name: 'select' },
+    description: 'Button icon',
+    table: {
+      type: { summary: 'string' },
+      defaultValue: { summary: '' },
+      category: 'Icon',
+    },
+    control: {
+      type: 'select',
+      options: iconsList,
+    },
+  };
+  argTypes.icon_transform = {
+    name: 'icon transform',
+    type: { name: 'select' },
+    description: 'Button icon transform',
+    table: {
+      type: { summary: 'string' },
+      defaultValue: { summary: '' },
+      category: 'Icon',
+    },
+    control: {
+      type: 'select',
+      options: [
+        'rotate-90',
+        'rotate-180',
+        'rotate-270',
+        'flip-horizontal',
+        'flip-vertical',
+      ],
+    },
+  };
+  argTypes.icon_position = {
+    name: 'icon position',
+    type: { name: 'inline-radio' },
+    description: 'Icon position inside the button',
+    table: {
+      type: { summary: 'string' },
+      defaultValue: { summary: 'after' },
+      category: 'Icon',
+    },
+    control: {
+      type: 'inline-radio',
+      options: ['before', 'after'],
+    },
+  };
+  argTypes.disabled = {
+    name: 'disabled',
+    type: { name: 'boolean' },
+    defaultValue: false,
+    description: 'Disabled button',
+    table: {
+      type: { summary: 'boolean' },
+      defaultValue: { summary: false },
+      category: 'States',
+    },
+    control: {
+      type: 'boolean',
+    },
+  };
+
+  return argTypes;
+};
 
-  getExtraKnobs(data);
+const applySpecs = (data, args) => {
```

Comment:
With apologies for the change of mind, but this function is `prepareData()` rather than applying specs. I didn't pay attention in an earlier comment what's happening in the function.